route.go

```diff
@@ -532,6 +532,30 @@ func (r *Route) URLPath(pairs ...string) (*url.URL, error) {
 	}, nil
 }
 
+// GetPathTemplate and GetHostTemplate returns the template used to match against for the route
+// This is userful for building simple REST API documentation,
+// and instrumentation for services like New Relic to ensure consistent reporting
+// The route must have a path defined.
+func (r *Route) GetPathTemplate() (string, error) {
+	if r.err != nil {
+		return "", r.err
+	}
+	if r.regexp == nil || r.regexp.path == nil {
+		return "", errors.New("mux: route doesn't have a path")
+	}
+	return r.regexp.path.template, nil
+}
+
+func (r *Route) GetHostTemplate() (string, error) {
```

Comment:
Need its own godoc string.

```go
```go
// GetHotsTemplate returns the template used to build the
// route match.
// An error will be returned if the route does not define a host.
```